test_huggingface_mae.py

```diff
@@ -0,0 +1,50 @@
+from huggingface_mae import MAEConfig, MAEModel
+
+config = MAEConfig(
+    input_norm={
+        "_target_": "torch.nn.Sequential",
+        "_args_": [
+            {"_target_": "normalizer.Normalizer"},
+            {"_target_": "torch.nn.LazyInstanceNorm2d", "affine": False, "track_running_stats": False}
+        ]
+    },
+    encoder={
+        "_target_": "mae_modules.MAEEncoder",
+        "channel_agnostic": True,
+        "max_in_chans": 11,
+        "vit_backbone": {
+            "_target_": "vit.sincos_positional_encoding_vit",
+            "vit_backbone": {
+                "_target_": "vit.vit_small_patch16_256",
+                "global_pool": "avg"
+            }
+        }
+    },
+    decoder={
+        "_target_": "mae_modules.CAMAEDecoder",
+        "num_modalities": 6,
+        "tokens_per_modality": 256,
+        "embed_dim": 256,
+        "depth": 4,
+        "num_heads": 16,
+        "mlp_ratio": 4,
+        "qkv_bias": True,
+        "norm_layer": {"_target_": "torch.nn.LayerNorm", "_partial_": True, "eps": 1e-6}
+    },
+    loss={"_target_": "torch.nn.MSELoss", "reduction": "none"},
+    fourier_loss={
+        "_target_": "loss.FourierLoss",
+        "num_multimodal_modalities": 6
+    }
+)
+
+# model = MAEModel.from_pretrained("models/phenom_beta", filename="last.pickle", from_state_dict=True)
+model = MAEModel.from_pretrained("models/phenom_beta", filename="last.pickle")
+
+huggingface_modelpath="recursionpharma/test-pb-model"
+model.push_to_hub(huggingface_modelpath)
+# model.save_pretrained(huggingface_modelpath, push_to_hub=True, repo_id=huggingface_modelpath)
+
+localpath = "models/phenom_beta_huggingface"
+model.save_pretrained(localpath)
+model = MAEModel.from_pretrained(localpath)
```

Comment:
is that localpath the `.safetensors` file?

if so we should add some additional testing logic here to sanity check that the model gives the same outputs when loaded from the `last.pickle` versus when loaded from hugging face.

the inference logic to test is thus:
```
def encoder_test_inference(model):
    with torch.inference_mode():
        x = model.example_input_array.clone()
        x = model.input_norm(x)
        embeddings = model.encoder(x)
    return embeddings
```

then we should do something like this:
```
model_last = MAEModel.from_pretrained("models/phenom_beta", filename="last.pickle")
embeddings_expected = encoder_test_inference(model_last)

model_hugging_face = MAEModel.from_pretrained(localpath)
embeddings_obtained = encoder_test_inference(model_hugging_face)

torch.testing.assert_close(embeddings_expected, embeddings_obtained)
```

Reply:
done